### PR TITLE
COP-2652: E2E test cases for adding people into sGMR user account

### PR DESCRIPTION
**Pull request by Bothees**
## Description
This Pull Request created for adding e2e test cases for adding people into user account. Test cases includes different scenarios ( both happy & un-happy path).

### How to run the test?
npm run cypress:test -- -- spec cypress/integration/sGMR/add-people.spec.js

## Developer Checklist

\* Required

- [x] Up-to-date with master branch? \*

- [x] Does it have tests?

- [x] Pull request URL linked to JIRA ticket? \*

- [ ] All reviewer comments resolved/answered? \*

Ensure you delete the branch once the Pull Request is merged.

### CODE REVIEW

File: cypress/support/utils.js [hunk 1]
```diff
@@ -10,3 +10,7 @@ exports.randomNumber = () => {
   const id = uuid();
   return id;
 };
+
+exports.getFutureDate = () => {
+  return Cypress.moment().add(2, 'year').format('DD/MM/YYYY');
```
Comment (p3dr0migue1) [inline cypress/support/utils.js:15]: This is good, but you could have also have used the faker library they also allow you to add a date in the future and past https://www.npmjs.com/package/faker

**Review by p3dr0migue1 — CHANGES_REQUESTED** [review]

**Comment by p3dr0migue1** [comment]
The tests are currently failing for me, I'll send you screenshots

File: cypress/support/utils.js (hunk 1 above)
Comment (Bothees) [reply cypress/support/utils.js]: I have checked the faker date generation, It generated Dates with specific format, like "2022-02-09T18:12:15.068Z".
We need the date in format "DD-MM-YYYY" so moment() has the ability to generate dates in any format.

File: cypress/integration/sGMR/add-people.spec.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    cy.url().should('include', '/people');
+    cy.get('table').getTable().then((peopleData) => {
+      expect(peopleData).to.not.be.empty;
+      cy.log(peopleData);
+      expectedPeople.forEach((item) => expect(peopleData).to.deep.include(item));
+    });
+  });
+
+  it('Should not add people without submitting required data', () => {
+    let errors = ['You must enter a first name',
```
[the remark this answers is not in the document]
Comment (Bothees) [reply cypress/integration/sGMR/add-people.spec.js]: done.

File: cypress/integration/sGMR/add-people.spec.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { randomNumber, getFutureDate } from '../../support/utils';
+
+describe('Add People in account', () => {
+  let PEOPLE;
+
+  before(() => {
+    cy.fixture('people.json').then((people) => {
+      people.DocumentNumber = randomNumber();
+      people.FirstName = `${people.FirstName}${randomNumber()}`;
```
[the remark this answers is not in the document]
Comment (Bothees) [reply cypress/integration/sGMR/add-people.spec.js]: Done. Used faker to generate Firstname & Lastname.

File: cypress/integration/sGMR/add-people.spec.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import { randomNumber, getFutureDate } from '../../support/utils';
+
+describe('Add People in account', () => {
+  let PEOPLE;
+
+  before(() => {
+    cy.fixture('people.json').then((people) => {
```
[the remark this answers is not in the document]
Comment (Bothees) [reply cypress/integration/sGMR/add-people.spec.js]: done.

File: cypress/integration/sGMR/add-people.spec.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import { randomNumber, getFutureDate } from '../../support/utils';
+
+describe('Add People in account', () => {
+  let PEOPLE;
```
[the remark this answers is not in the document]
Comment (Bothees) [reply cypress/integration/sGMR/add-people.spec.js]: done.

File: cypress/fixtures/people.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "FirstName": "test",
```
[the remark this answers is not in the document]
Comment (Bothees) [reply cypress/fixtures/people.json]: removed the hardcoded name.

**Review by p3dr0migue1 — DISMISSED** [review]

**Review by p3dr0migue1 — APPROVED** [review]

**Pull request merged**